install.sh

```diff
@@ -56,8 +56,12 @@ if ! sudo apt-get update; then
     die_cmd "Failed to update APT cache."
 fi
 
-if ! sudo apt-get install gpg wget; then
-    die_cmd "Failed to check if needed packages are installed."
+missing_dependencies=()
+dpkg-query -W 'wget' > /dev/null 2>&1 || missing_dependencies+=('wget')
+dpkg-query -W 'gpg' > /dev/null 2>&1 || missing_dependencies+=('gpg')
+
+if ! ( test -z "${missing_dependencies[*]}" || sudo apt-get install --mark-auto "${missing_dependencies[*]}" ); then
```

Comment:
The `sudo apt-get install --mark-auto "${missing_dependencies[*]}"` needs to be ` "${missing_dependencies[@]}"` so that each item in the array is expanded into a separate argument though. The `*` makes it all go into one string, while the `@` makes them all separate strings that are passed to the command.

Reply:
If you run your current code locally you'll see that it doesn't work, as it looks for a single package named `wget gpg` if both aren't installed.

Reply:
> you misdiagnosed the problem

How so? It looks like your method for finding packages that aren't installed worked just fine, it's just that the Bash syntax with the `[*]` stuff didn't look correct. Was that not how it behaved for you?